docs/DeveloperGuide.md

```diff
@@ -1,38 +1,442 @@
+
+2. **Set up CI** This project comes with a GitHub Actions config files (in .github/workflows folder). When GitHub detects those files, it will run the CI for your project automatically at each push to the master branch or to any PR. No set up required.  
+3. **Learn the design** When you are ready to start coding, we recommend that you get some sense of the overall design by reading about [AddressBook’s architecture](https://se-education.org/addressbook-level3/DeveloperGuide.html#architecture).  
+4. **Do the tutorials These tutorials will help you get acquainted with the codebase.**  
+    * [Tracing code](https://se-education.org/addressbook-level3/tutorials/TracingCode.html)
+    * [Adding a new command](https://se-education.org/addressbook-level3/tutorials/AddRemark.html)
+    * [Removing fields](https://se-education.org/addressbook-level3/tutorials/RemovingFields.html)
+
+
+---
+
+[//]: # (@@author ollayf)
+# Design & Implementation
+
+
+## Architecture
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ArchitectureDiagram.png" width="100%" />
+
+
+This **Architecture Diagram** explains the high-level design of the App.
+
+
+### Main Components of the architecture
+
+**SecureNUS** has 1 class called SecureNUS. It is responsible for,
+* At app launch: initializing all the important components in the correct sequence and initializes the appropriate connections between them.
+* It connects and sends the appropriate information and executes different actions using the other helpers via **Commands**.
+
+**SecretStorage** is responsible for the management of Secret Objects which store information of different Secret classes, e.g. passwords, credit cards. It also manages the different underlying data structures within it that help optimize the speed of queries through indexing, etc. It consists of 3 classes: **SecretMaster, SecretSearcher** and **SecretEnumerator**, which contain the different views.
+
+
+
+* **SecretMaster:** Manages the other views and also ensures there are no duplicates in naming where they should not exist
+* **SecretSearcher:** Contains the Hashtable view for directly querying the Secret by name
+* **SecretEnumerator:** Contains the ArrayList view for querying the Secret by index
+
+The other 2 components are:
+
+
+
+* **UI:** The main interface the user have access to through the Command Line
+* **Backend:** Which handles the details regarding the initialisation and saving (or exporting) of data across sessions
+
+<img src="./DGDiagramsCreator/DGUsedDiagrams/ArchitectureDiagram.png" width="100%" />  
+
+___
+### How the architecture components interact with each other
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user creates a new basic password initiated using the command `new password.`
+
+Sequence Diagram
```

Comment:
In the sequence diagram, a semicolon should be add before the class name. Unnamed instances of a class should be denoted as `:Class`.